File: frontend/src/lib/types/budget.ts
export type BudgetStatus = 0 | 1 | 2 | 3; // 0=disabled, 1=active, 2=completed, 3=overrun
export type RiskLevel = "low" | "medium" | "high" | "critical";

export interface Budget {
  id: number;
  version?: number;
  name: string;
  amount: number;
  startDate: string;
  endDate: string;
  warningThreshold: number;
  status: BudgetStatus;
  spent: number;
  remainingAmount: number;
  usageRate: number;
  warningReached: boolean;
  riskLevel: RiskLevel;
  riskMessage: string;
  userId: number;
  ledgerId: number | null;
  categoryId: number | null;
  categoryName?: string;
  categoryIcon?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateBudgetDTO {
  name: string;
  amount: number;
  startDate: string;
  endDate: string;
  warningThreshold: number;
  categoryId?: number;
}

export interface UpdateBudgetDTO {
  version: number;
  name?: string;
  amount?: number;
  startDate?: string;
  endDate?: string;
  warningThreshold?: number;
  categoryId?: number;
  clearCategory?: boolean;
  status?: BudgetStatus;
}

export interface BudgetQuery {
  status?: BudgetStatus;
  startDate?: string;
  endDate?: string;
  keyword?: string;
  page?: number;
  size?: number;
}

File: backend/src/main/java/com/mamoji/budget/domain/Budget.java
package com.mamoji.budget.domain;

import java.math.BigDecimal;

/** Budget-owned capacity, usage, and category projection. */
public class Budget {
    public long id;
    public long version;
    public Long companyId;
    public String name;
    public BigDecimal amount;
    public String startDate;
    public String endDate;
    public int warningThreshold;
    public int status;
    public BigDecimal spent;
    public BigDecimal reservedAmount;
    public BigDecimal remainingAmount;
    public BigDecimal availableAmount;
    public double usageRate;
    public boolean warningReached;
    public String riskLevel;
    public String riskMessage;
    public long userId;
    public Long ledgerId;
    public Long categoryId;
    public String categoryName;
    public String categoryIcon;
    public String createdAt;
    public String updatedAt;
}
